drake/core/Vector.h

```diff
@@ -0,0 +1,269 @@
+
+#include <string>
+#include <vector>
+#include <stdexcept>
+#include <iostream>
+#include <Eigen/Dense>
+
+namespace Drake {
+
+  /** @defgroup vector_concept Vector<ScalarType> Concept
+   * @ingroup concepts
+   * @{
+   * @brief Describes a (potentially structured) data which can be operated on as a (finite-dimensional) column vector
+   *
+   * @nbsp
+   *
+   * | Valid Expressions (which must be implemented) |  |
+   * ------------------|-------------------------------------------------------------|
+   * | RowsAtCompileTime  | defined as a static constant int (or enum).  Can be Eigen::Dynamic. |
+   * | size_t size()      | only required if RowsAtCompileTime==Eigen::Dynamic |
+   * | template<Derived> Vector(const Eigen::MatrixBase<Derived>&)  | constructor taking an Eigen object |
+   * | template<Derived> Vector& operator=(const Eigen::MatrixBase<Derived>&)   | assignment operator from an Eigen object |
+   * | Eigen::Matrix<ScalarType,RowsAtCompileTime,1> toEigen(const Vector<ScalarType>&) | non-member namespace method which converts to the Eigen type |
+   *
+   * @}
+   */
+
+  /** EigenVector<Rows>::type<ScalarType>
+   * @brief provides an alias for Eigen::Matrix<ScalarType,Rows,1> which is templated on only a single argument (the ScalarType)
+   * @concept{vector_concept}
+   */
+  template<int Rows>
+  struct EigenVector {
+    template<typename ScalarType> using type = Eigen::Matrix<ScalarType, Rows, 1>;
+  };
+
+  typedef Eigen::Matrix<double,1,1> Vector1d;
+
+  /** NullVector<ScalarType>
+   * @brief provides the empty vector (templated by ScalarType)
+   * @concept{vector_concept}
+   */
+  template<typename ScalarType> using NullVector = Eigen::Matrix<ScalarType, 0, 1>;
+
+
+  template<typename ScalarType, int Rows>
+  Eigen::Matrix<ScalarType, Rows, 1> toEigen(const Eigen::Matrix<ScalarType, Rows, 1> &vec) { return vec; }
```

Comment:
Would be nice if we could return a reference from this to avoid the copy. Is that safe?